ios/Codeforces Watcher/Application/features/users/cells/userAccount/UserAccountPerformanceView.swift

```diff
@@ -0,0 +1,249 @@
+        [contributionIcon, contributionLabel].forEach(contributionView.addSubview)
+    }
+
+    private func setConstraints() {
+        contentView.edgesToSuperview(insets: UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 8))
+        
+        ratingView.edgesToSuperview(excluding: .bottom)
+        
+        ratingIcon.run {
+            $0.width(15)
+            $0.height(15)
+            $0.edgesToSuperview(excluding: .trailing)
+        }
+        
+        ratingLabel.run {
+            $0.verticalToSuperview()
+            $0.leadingToTrailing(of: ratingIcon, offset: 4)
+        }
+        
+        maxRatingView.run {
+            $0.topToBottom(of: ratingView, offset: 8)
+            $0.leadingToSuperview(offset: 2.5)
+            $0.trailingToSuperview()
+        }
+        
+        maxRatingIcon.run {
+            $0.width(11)
+            $0.height(11)
+            $0.edgesToSuperview(excluding: .trailing)
+        }
+        
+        maxRatingLabel.run {
+            $0.verticalToSuperview()
+            $0.leadingToTrailing(of: ratingIcon, offset: 3.5)
+        }
+        
+        contributionView.run {
+            $0.topToBottom(of: maxRatingView, offset: 5)
+            $0.horizontalToSuperview()
+        }
+        
+        contributionIcon.run {
+            $0.width(16)
+            $0.height(16)
+            $0.edgesToSuperview(excluding: .trailing)
+        }
+        
+        contributionLabel.run {
+            $0.topToSuperview(offset: 3)
+            $0.bottomToSuperview()
+            $0.leadingToTrailing(of: ratingIcon, offset: 4)
+        }
+        
+        lineChartView.run {
+            $0.topToBottom(of: contributionIcon, offset: 4)
+            $0.edgesToSuperview(excluding: .top)
+        }
+    }
+    
+    func bind(_ uiModel: UserAccountPerformanceView.UIModel) {
+        self.uiModel = uiModel
+        
+        ratingLabel.attributedText = uiModel.ratingText
+        maxRatingLabel.attributedText = uiModel.maxRatingText
+        contributionLabel.attributedText = uiModel.contributionText
+        
+        displayChart()
+    }
+    
+    private func displayChart() {
+        guard let ratingChanges = uiModel?.ratingChanges else { return }
+        
+        let dataEntries = ratingChanges.map {
+            ChartDataEntry(
+                x: Double($0.ratingUpdateTimeSeconds),
+                y: Double($0.newRating),
+                data: $0.toChartItem()
+            )
+        }
+        
+        let dataSet = LineChartDataSet(entries: dataEntries).apply {
+            $0.lineWidth = 1.5
+            $0.circleRadius = 2.5
+            $0.circleHoleRadius = 1.5
+            $0.circleHoleColor = Palette.white
+        }
+        
+        let data = LineChartData(dataSet: dataSet).apply {
+            $0.setDrawValues(false)
+        }
+        
+        lineChartView.data = data
+    }
+}
+
+fileprivate extension UserAccountPerformanceView.UIModel {
+    
```

Comment:
Looks like this extension get repeated in several files so you can extract it to separate file

Reply:
I cannot because these methods are quite different from each other.